### PR TITLE
tests: add test infra and one test

### PR DESCRIPTION
**Pull request by maelle**

### CODE REVIEW

File: R/lazytest_local.R [hunk 1]
```diff
@@ -69,7 +69,7 @@ lazytest_local <- function(path = ".",
     )
   }
 
-  out <- test_local(
+  out <- testthat::test_local(
```
Comment (maelle) [inline R/lazytest_local.R:72]: otherwise callr doesn't find it

File: tests/testthat/test-lazytest_local.R [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  )
+  expect_equal(
+    executed_test_files(first_run),
+    c("test-blip.R", "test-blop.R")
+  )
+
+  second_run <- run_lazytest(pkg_dir = pkg_dir, lazytest_dir = lazytest_dir())
+  expect_equal(
+    executed_test_files(second_run),
+    c("test-blip.R")
```
Comment (maelle) [inline tests/testthat/test-lazytest_local.R:33]: only failed test run

**Review by krlmlr — APPROVED** [review]
Can you please skip the failing tests on non-UTF-8 platforms? Looks great otherwise!

File: tests/testthat/test-lazytest_local.R [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  )
+
+  # fix failing test, run tests twice ----
+  edit_test("blip", passing_test_lines(), pkg_dir = pkg_dir)
+  new_run <- run_lazytest(pkg_dir = pkg_dir, lazytest_dir = lazytest_dir())
+  expect_equal(
+    executed_test_files(new_run),
+    c("test-blip.R", "test-blop.R")
```
[the remark this answers is not in the document]
Comment (maelle) [reply tests/testthat/test-lazytest_local.R:42]: Aaaah this makes sense, thanks!

**Comment by maelle** [comment]
@krlmlr I can't merge this since I don't have write access :wink:

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**